components/google-cloud/google_cloud_components/mb_sdk_component/utils.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for creating pipeline components based on AI Platform SDK."""
+
+import inspect
+import kfp
+from kfp import components
+
+INIT_KEY = "init"
+METHOD_KEY = "method"
+
+
+def convert_method_to_component(method, should_serialize_init=False):
+  """Creates a corresponding KFP Component for a given MBD SDK method."""
+  method_name = method.__name__
+
+  if inspect.ismethod(method):
+    cls_name = method.__self__.__name__
+    init_signature = inspect.signature(method.__self__.__init__)
+  else:
+    cls = getattr(
+        inspect.getmodule(method),
+        method.__qualname__.split(".<locals>", 1)[0].rsplit(".", 1)[0], None)
+    cls_name = cls.__name__
+    init_signature = inspect.signature(cls.__init__)
+
+  init_arg_names = set(
+      init_signature.parameters.keys()) if should_serialize_init else set([])
+
+  def make_args(sa):
+    additional_args = []
+    for key, args in sa.items():
+      for arg_key, value in args.items():
+        additional_args.append(f"    - --{key}.{arg_key}={value}")
+    return "\n".join(additional_args)
+
+  def f(**kwargs):
```

Comment:
Can this have a more descriptive name and docs?

Reply:
done, renamed to component_yaml_generator and added short descriptive doc string.